image.go

```diff
@@ -43,7 +43,7 @@ const (
 	Dockerv2ImageManifest = define.Dockerv2ImageManifest
 )
 
-type containerImageRef struct {
+type ContainerImageRef struct {
```

Comment:
We can move `ContainerImageRef` to internal but as of now I don't see lot of harm in making it public since it has access to some useful methods and ContainerImageRef does not changes frequently but making it internal is safer.

Reply:
(In the case of this one in particular, the last revision of the PR I saw needed to be able to call `ExtractRootFS` from a different package, and I didn't have a better way to accomplish that in-hand.)

Reply:
Okay, an alternative would be to leave the `containerImageRef` type and its `extractRootfs()` method private, and add a public method to `Builder` that internally calls `makeContainerImageRef()` to create a `containerImageRef`, calls the resulting ref's `extractRootfs()` method, and hands the result back to the new methods's caller.  That's enough for the use case in `StageExecutor.commit()`, and doesn't expose a lot of our internals, so I think I like it better.

Reply:
ackd SGTM I'll add these changes.

Reply:
fixed.